src/main/java/arsenic/injection/mixin/MixinEntityPlayer.java

```diff
@@ -1,20 +1,187 @@
+                float f = (float) this.getEntityAttribute(SharedMonsterAttributes.attackDamage).getAttributeValue();
+                float f1 = 0.0F;
+                if (p_attackTargetEntityWithCurrentItem_1_ instanceof EntityLivingBase) {
+                    f1 = EnchantmentHelper.func_152377_a(this.getHeldItem(),
+                            ((EntityLivingBase) p_attackTargetEntityWithCurrentItem_1_).getCreatureAttribute());
+                } else {
+                    f1 = EnchantmentHelper.func_152377_a(this.getHeldItem(), EnumCreatureAttribute.UNDEFINED);
+                }
+
+                int i = EnchantmentHelper.getKnockbackModifier(this);
+                if (this.isSprinting()) {
+                    ++i;
+                }
+
+                if (f > 0.0F || f1 > 0.0F) {
+                    boolean flag = this.fallDistance > 0.0F && !this.onGround && !this.isOnLadder() && !this.isInWater()
+                            && !this.isPotionActive(Potion.blindness) && this.ridingEntity == null
+                            && p_attackTargetEntityWithCurrentItem_1_ instanceof EntityLivingBase;
+                    if (flag && f > 0.0F) {
+                        f *= 1.5F;
+                    }
+
+                    f += f1;
+                    boolean flag1 = false;
+                    int j = EnchantmentHelper.getFireAspectModifier(this);
+                    if (p_attackTargetEntityWithCurrentItem_1_ instanceof EntityLivingBase && j > 0
+                            && !p_attackTargetEntityWithCurrentItem_1_.isBurning()) {
+                        flag1 = true;
+                        p_attackTargetEntityWithCurrentItem_1_.setFire(1);
+                    }
+
+                    double d0 = p_attackTargetEntityWithCurrentItem_1_.motionX;
+                    double d1 = p_attackTargetEntityWithCurrentItem_1_.motionY;
+                    double d2 = p_attackTargetEntityWithCurrentItem_1_.motionZ;
+                    boolean flag2 = p_attackTargetEntityWithCurrentItem_1_
+                            .attackEntityFrom(DamageSource.causePlayerDamage(((EntityPlayer) (Object) this)), f);
+                    if (flag2) {
+                        if (i > 0) {
+                            p_attackTargetEntityWithCurrentItem_1_.addVelocity(
+                                    -MathHelper.sin(this.rotationYaw * 3.1415927F / 180.0F) * (float) i * 0.5F, 0.1D,
+                                    MathHelper.cos(this.rotationYaw * 3.1415927F / 180.0F) * (float) i * 0.5F);
+
+                            KeepSprint keepSprint = Arsenic.getArsenic().getModuleManager().getModuleByClass(KeepSprint.class);
+                            if (keepSprint.isEnabled()) {
+                                this.motionX *= keepSprint.value.getValue().getInput();
+                                this.motionZ *= keepSprint.value.getValue().getInput();
+                                this.setSprinting(keepSprint.sprint.getValue());
```

Comment:
inject an event over here and use it from keepsprint class and modify player motion there.
Pasting an entire method to modify a few lines is a bad idea
also inject attack event at head as well (you removed it for some reason)

Reply:
![image](https://github.com/ArsenicClient/Arsenic/assets/123317160/85e80ff5-5eb6-410e-a496-8719a58693ef)
no?

Reply:
do the first thing